src/app/file-explorer.js

```diff
@@ -251,9 +239,9 @@ function fileExplorer (appAPI, files) {
     element = el
   }
 
-  element.events = events
-  element.api = api
-  return element
+  api.events = events
+  container.appendChild(element)
+  return api
```

Comment:
for `3.` not really:

```
// 3. i understand your above proposal as:
var foobar = {
  events: new EventManager(),
  api: {
    callAFoobarMethod: function () {},
    callAnotherMethod: function () {}
  }
}
```

The api object contains all the stuff that the module require from the outside.
And the way it is instanciated is: 
for instance for the Debugger:

var debug = new debug("#domid", api, executionContext.event, edtior.event)
the debug component needs this `api` to retrieve stuff from the outside, and need to register to some events. In that case he'll register to the `editor event` and `execution context event`
e.g https://github.com/ethereum/browser-solidity/blob/master/src/app.js#L650

Note that the code of the dom component (yoyo) is completely separated from the execution logic.